pkg/storegateway/bucket.go

```diff
@@ -1130,6 +1130,7 @@ func (s *BucketStore) streamingSeriesSetForBlocks(
 				matchers,
 				shardSelector,
 				blockSeriesHashCache,
+				cachedSeriesHasher{blockSeriesHashCache},
```

Comment:
`filterPostingsByCachedShardHash` can only look at the cache. I kept them as two so that I don't change the default implementation.

This commit makes them one https://github.com/grafana/mimir/pull/3728/commits/7453753ee6247d78062cf97f51b1906ee7920e9c and changes the default implementaiton